core/domain/state_domain.py

```diff
@@ -916,8 +940,1085 @@ def get_all_outcomes(self) -> List[Outcome]:
+                            f'value - {rule_value}'
+                        )
+
+                if rule_spec.rule_type == 'IsWithinTolerance':
+                    rule_value_x = rule_spec.inputs['x']
+                    rule_value_tol = rule_spec.inputs['tol']
+                    try:
+                        rule_value_x = float(rule_value_x)
+                        rule_value_tol = float(rule_value_tol)
+                        # `tol` value in `IsWithinTolerance` must be positive.
+                        if rule_value_tol <= 0.0:
+                            raise utils.ValidationError(
+                                f'The rule \'{rule_spec_index}\' of answer '
+                                f'group \'{ans_group_index}\' having '
+                                f'rule type \'IsWithinTolerance\' '
+                                f'have \'tol\' value less than or equal to '
+                                f'zero in NumericInput interaction.'
+                            )
+                        self._set_lower_and_upper_bounds(
+                            range_var, rule_value_x - rule_value_tol,
+                            rule_value_x + rule_value_tol, True, True
+                        )
+                    except Exception:
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' having rule '
+                            f'type \'IsWithinTolerance\' contains string '
+                            f'value - \'{rule_value_x}\', \'{rule_value_tol}\''
+                        )
+
+                if rule_spec.rule_type == 'IsGreaterThan':
+                    try:
+                        rule_value = float(rule_spec.inputs['x'])
+                        self._set_lower_and_upper_bounds(
+                            range_var, rule_value,
+                            upper_infinity, False, False
+                        )
+                    except Exception:
+                        rule_value = rule_spec.inputs['x']
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' having rule '
+                            f'type \'IsGreaterThan\' contains string '
+                            f'value - \'{rule_value}\''
+                        )
+
+                if rule_spec.rule_type == 'IsInclusivelyBetween':
+                    rule_value_a = rule_spec.inputs['a']
+                    rule_value_b = rule_spec.inputs['b']
+                    try:
+                        rule_value_a = float(rule_value_a)
+                        rule_value_b = float(rule_value_b)
+                        # `a` should not be greater than `b` in
+                        # `IsInclusivelyBetween` rule.
+                        if rule_value_a > rule_value_b:
```

Comment:
Did we decide that `rule_value_a == rule_value_b` is okay and valid?

Reply:
Nope, I have updated this